app/src/main/java/com/github/se/polyfit/ui/screen/MapScreen.kt

```diff
@@ -24,88 +25,112 @@ import androidx.compose.ui.unit.dp
 import androidx.hilt.navigation.compose.hiltViewModel
 import com.github.se.polyfit.model.post.Location
 import com.github.se.polyfit.model.post.Post
+import com.github.se.polyfit.ui.components.postinfo.PostCard
 import com.github.se.polyfit.ui.theme.PrimaryPurple
 import com.github.se.polyfit.viewmodel.map.MapViewModel
+import com.google.android.gms.maps.model.BitmapDescriptorFactory
 import com.google.android.gms.maps.model.CameraPosition
 import com.google.android.gms.maps.model.LatLng
 import com.google.maps.android.compose.Circle
 import com.google.maps.android.compose.GoogleMap
+import com.google.maps.android.compose.GoogleMapComposable
 import com.google.maps.android.compose.MapProperties
 import com.google.maps.android.compose.Marker
 import com.google.maps.android.compose.MarkerState
 import com.google.maps.android.compose.rememberCameraPositionState
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.FlowPreview
 import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
 import kotlinx.coroutines.sync.Mutex
 import kotlinx.coroutines.sync.withLock
 
 // Size in meters of the rayon of the circle to see other posts
-val MIN_SIZE_RAYON = 100f
-val MAX_SIZE_RAYON = 2000f
-val EPFL_LATITUDE = 46.5181
-val EPFL_LONGITUDE = 6.5659
+const val MIN_SIZE_RAYON = 100f
+const val MAX_SIZE_RAYON = 2000f
+const val EPFL_LATITUDE = 46.5181
+const val EPFL_LONGITUDE = 6.5659
+const val THREE_SECONDS = 3000L
 
+@OptIn(FlowPreview::class)
 @Composable
-fun MapScreen(
-    paddingValues: PaddingValues,
-    goToPost: () -> Unit,
-    mapviewMap: MapViewModel = hiltViewModel()
-) {
+fun MapScreen(paddingValues: PaddingValues, mapViewModel: MapViewModel = hiltViewModel()) {
 
   val posts = remember { mutableStateOf(listOf<Post>()) }
+
   val currentLocation = remember { mutableStateOf(Location.default()) }
-  val mutex = Mutex()
+
```

Comment:
It might be a good initiative to use viewModel states to declare these values and collect them as state in the UI but it isn't that important at the moment, could be modified at a later notice